### PR TITLE
post: ng perf series part 1

### PR DESCRIPTION
**Pull request by cexbrayat**
Scheduled for Thursday if you can review it before.
It's mainly the content of the ebook, reformated in Markdown,
with an intro and outro.

### CODE REVIEW

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+We have just finished a new chapter of our [ebook](https://books.ninja-squad.com/angular) about performances,
+and we thought we could share it with you in a series of blog posts.
+It took us a long time, but we wanted to write something more complete than what you can usually find.
+There are a lot of tips to make Angular faster (whatever faster means for you, we'll come back to this in a minute),
+but you usually don't have the other side of the story: what are the traps of these optimizations,
+are they what you are looking for,
+and should you really use them.
+
+This is the first part of this series, and this blog post is about the first laod of an Angular application.
```
Comment (clacote) [inline _posts/2018-09-06-angular-performances-part-1.md]: "series", like a plural?

Comment (clacote) [inline _posts/2018-09-06-angular-performances-part-1.md]: laod -> load

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+In future posts, we'll talk about how to make reloading faster, then about how to profile your running application,
+and who to improve runtime performances.
```
Comment (clacote) [inline _posts/2018-09-06-angular-performances-part-1.md]: who -> what, no?

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+## First load
+
+When you load a modern Web application in your browser, a few things happen. First, the `index.html` is loaded and parsed by the browser. Then the JS scripts and other assets referenced are fetched. When one of the assets is received, the browser parses it, and executes it if it is a JS file.
+
+### Assets sizes
+
+So the first tip is very obvious: be careful with your assets sizes!
+
+The assets loading phase depends on how many assets you want to load. A lot will be slow. Big ones will be slow. Especially if the network is not that good, which happens more than you think: you might test your application of a fiber optic connection, but some of your actual users might be in the middle of nowhere, using slow 3G. Here is what you can do.
```
Comment (clacote) [inline _posts/2018-09-06-angular-performances-part-1.md]: test your application of a fiber optic connection -> on a connection, or over a connection

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+---
+layout: post
+title: Angular Performances Part 1 - First load
+author: cexbrayat
+tags: ["Angular 6", "Angular 5", "Angular", "Angular 2", "Angular 4", "Angular CLI"]
```
Comment (clacote) [inline _posts/2018-09-06-angular-performances-part-1.md]: I would add a first tag like "performances" to link the complete series. And maybe "benchmark" for SEO.

File: _posts/2018-09-06-angular-performances-part-1.md (hunk 1 above)
Comment (cexbrayat) [reply _posts/2018-09-06-angular-performances-part-1.md]: I think "serie" doesn't exist but maybe I'm wrong...

File: _posts/2018-09-06-angular-performances-part-1.md (hunk 2 above)
Comment (cexbrayat) [reply _posts/2018-09-06-angular-performances-part-1.md]: was how 😅

File: _posts/2018-09-06-angular-performances-part-1.md (hunk 3 above)
Comment (cexbrayat) [reply _posts/2018-09-06-angular-performances-part-1.md]: 👍 I'll fix in ebook too.

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+It took us a long time, but we wanted to write something more complete than what you can usually find.
+There are a lot of tips to make Angular faster (whatever faster means for you, we'll come back to this in a minute),
+but you usually don't have the other side of the story: what are the traps of these optimizations,
+are they what you are looking for,
+and should you really use them.
+
+This is the first part of this series, and this blog post is about the first load of an Angular application.
+In future posts, we'll talk about how to make reloading faster, then about how to profile your running application,
+and how to improve runtime performances.
+If you are the lucky owner of our ebook, you can already check the rest if you download the [last ebook release](https://books.ninja-squad.com/claim?book=Angular).
```
Comment (jnizet) [inline _posts/2018-09-06-angular-performances-part-1.md]: the rest -> the other parts

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+## First load
+
+When you load a modern Web application in your browser, a few things happen. First, the `index.html` is loaded and parsed by the browser. Then the JS scripts and other assets referenced are fetched. When one of the assets is received, the browser parses it, and executes it if it is a JS file.
+
+### Assets sizes
+
+So the first tip is very obvious: be careful with your assets sizes!
+
+The assets loading phase depends on how many assets you want to load. A lot will be slow. Big ones will be slow. Especially if the network is not that good, which happens more than you think: you might test your application on a fiber optic connection, but some of your actual users might be in the middle of nowhere, using slow 3G. Here is what you can do.
```
Comment (jnizet) [inline _posts/2018-09-06-angular-performances-part-1.md]: happens more -> happens more often

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+The bad news is that it’s not as easy as adding the `@angular/universal` package. You application needs to follow some best practices (no direct DOM manipulation for example, as the server won’t have a real DOM to manipulate). Then you need to setup your server and think about the strategy you want to adopt. Do you want to pre-render all pages or just a few? Do you want to pre-render the whole page, with the data fetching and authorization check it will need, or just some critical parts of the page? Do you want to pre-render them on build, or to pre-render them on demand and cache them? Do you want to do this for all the possible profiles and languages or just some? All these questions depends on the type of application you are building, and the effort can vary greatly depending on your goal.
+
+So, again, I would advise you to use server side rendering only if it is critical for your application, and not based on the hype…​
+
+If you enjoy this blog post, you may want to dig deeper with our [ebook](https://books.ninja-squad.com/angular),
```
Comment (jnizet) [inline _posts/2018-09-06-angular-performances-part-1.md]: enjoy -> enjoyed

File: _posts/2018-09-06-angular-performances-part-1.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+and/or with a complete exercise that we added in our [online training](https://angular-exercises.ninja-squad.com/).
+The exercise takes an application and walks you through what we would do to optimize it,
+measuring the benefits of each steps, showing you how to avoid the common traps,
+how to test the optimized application, etc. Check it out if you want to learn more!
+
+See you next for part 2.
```
Comment (jnizet) [inline _posts/2018-09-06-angular-performances-part-1.md]: See you soon

File: _posts/2018-09-06-angular-performances-part-1.md (hunk 6 above)
Comment (cexbrayat) [reply _posts/2018-09-06-angular-performances-part-1.md]: I'll fix in the ebook too.

**Pull request merged**